provider/agent_test.go

```diff
@@ -111,3 +111,49 @@ func TestAgentInstance(t *testing.T) {
 		}},
 	})
 }
+
+func TestAgent_Metadata(t *testing.T) {
+	t.Parallel()
+	resource.Test(t, resource.TestCase{
+		Providers: map[string]*schema.Provider{
+			"coder": provider.New(),
+		},
+		IsUnitTest: true,
+		Steps: []resource.TestStep{{
+			Config: `
+				provider "coder" {
+					url = "https://example.com"
+				}
+				resource "coder_agent" "dev" {
+					os = "linux"
+					arch = "amd64"
+					metadata {
+						key = "process_count"
```

Comment:
Thoughts on calling this `id` instead? It's consistent with other naming we've done in here.

Reply:
We can make it id... We don't really want people thinking they have to auto generate it though.